test/test_tune.py

```diff
@@ -203,3 +184,23 @@ def test_autotune_match(self, open_db_mock: mock.Mock, close_db_mock: mock.Mock,
             context.device.name = 'z'
             ret = self.autotune(context, 'xyz')
             assert ret == tuning_1 or ret == tuning_2
+
+            # manually reset autotune_mock to no longer raise RuntimeError
+            self.autotune_mock.side_effect = None
```

Comment:
Rather do that in `_prepare_autotune_mock`, so that it happens even if the test fails.